File: java/src/main/java/ch/frostnova/example/shoppingcart/model/Unit.java
package ch.frostnova.example.shoppingcart.model;

public enum Unit {

    QUANTITY("x"),
    KILOGRAM("kg"),
    LITER("l");

    private final String displayName;

    Unit(String displayName) {
        this.displayName = displayName;
    }

    @Override
    public String toString() {
        return displayName;
    }
}

File: rust/src/unit.rs
use std::fmt;

pub enum Unit {
    QUANTITY,
    KILOGRAM,
    LITER,
}

impl fmt::Display for Unit {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        match *self {
            Unit::QUANTITY => write!(f, "x"),
            Unit::KILOGRAM => write!(f, "kg"),
            Unit::LITER => write!(f, "l"),
        }
    }
}